plugin/trace/trace.go

```diff
@@ -81,34 +89,29 @@ func (t *trace) Name() string { return "trace" }
 
 // ServeDNS implements the plugin.Handle interface.
 func (t *trace) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
-	trace := false
-	if t.every > 0 {
-		queryNr := atomic.AddUint64(&t.count, 1)
-
-		if queryNr%t.every == 0 {
-			trace = true
-		}
-	}
 	span := ot.SpanFromContext(ctx)
-	if !trace || span != nil {
+	if span != nil {
 		return plugin.NextOrFailure(t.Name(), t.Next, ctx, w, r)
 	}
 
 	req := request.Request{W: w, Req: r}
-	span = t.Tracer().StartSpan(spanName(ctx, req))
+	span = t.Tracer().StartSpan("servedns")
 	defer span.Finish()
 
 	rw := dnstest.NewRecorder(w)
 	ctx = ot.ContextWithSpan(ctx, span)
 	status, err := plugin.NextOrFailure(t.Name(), t.Next, ctx, rw, r)
 
+	span.SetTag(tagClient, req.IP())
 	span.SetTag(tagName, req.Name())
-	span.SetTag(tagType, req.Type())
+	span.SetTag(tagProto, req.Proto())
 	span.SetTag(tagRcode, rcode.ToString(rw.Rcode))
+	span.SetTag(tagType, req.Type())
+	if t.EndpointType == "datadog" {
```

Comment:
Datadog APM has a concept of "Resource" while allows to aggregate stats at an extra level compared to the span name.
In SQL, we use a quantized resource.
In a web framework, we use the controller's name.

Here, would any dimension make sense? Are the served DNS easy to regrouped (in groups which make sense to be seen aggregated?).

Reply:
Once they've been analyzed the tags can be used to group top span easily. 

I don't think the span tree is complex enough to need more grouping.

Reply:
Would the queried domain make sense as a "Resource"?

Reply:
It may. How many different domains could we expect (within up to hundreds)?
Does it interest you to differentiate the performance of a domain vs. another?

Reply:
1. Yes: we're actually more interested in the performance for different zones because they are handled differently (cluster.local is served directly, service.consul / tina / . are forwarded to different upstream resolvers). What makes things "hard" to read is caching (so for all zones we expect a bi-modal distribution: served from cached / forwarded)
2. A lot (100s, easily)